include/filewriter/summarywriter_impl.hpp

```diff
@@ -22,6 +22,102 @@ void SummaryWriter<Filewriter>::Scalar(const std::string& tag,
+
+  mlboard::Summary_Image *image = new Summary::Image();
+  image->set_height(height);
+  image->set_width(width);
+  image->set_colorspace(channel);
+  image->set_encoded_image_string(encodedImage);
+
+  mlboard::Summary *summary = new Summary();
+  mlboard::Summary_Value *v = summary->add_value();
+  v->set_tag(tag);
+  v->set_allocated_image(image);
+  v->set_allocated_metadata(meta);
+
+  fw.CreateEvent(step, summary);
+}
+template<typename Filewriter>
+void SummaryWriter<Filewriter>::Image(const std::string& tag,
+                                      int step,
+                                      const std::vector<std::string>& 
+                                        encodedImages,
+                                      int height,
+                                      int width,
+                                      Filewriter& fw,
+                                      const std::string& displayName,
+                                      const std::string& description)
+{
+  mlboard::SummaryMetadata_PluginData *pluginData =
+    new SummaryMetadata::PluginData();
+  pluginData->set_plugin_name("images");
+  mlboard::SummaryMetadata *meta = new SummaryMetadata();
+  meta->set_display_name(displayName == "" ? tag : displayName);
+  meta->set_summary_description(description);
+  meta->set_allocated_plugin_data(pluginData);
+
+  mlboard::TensorProto *tensor = new TensorProto();
+  tensor->set_dtype(mlboard::DataType::DT_STRING);
+  tensor->add_string_val(std::to_string(width));
+  tensor->add_string_val(std::to_string(height));
+  for (const std::string& image: encodedImages)
+      tensor->add_string_val(image);
+
+  mlboard::Summary *summary = new Summary();
+  mlboard::Summary_Value *v = summary->add_value();
+  v->set_tag(tag);
+  v->set_allocated_tensor(tensor);
+  v->set_allocated_metadata(meta);
+  fw.CreateEvent(step, summary);
+}
+
+  template<typename Filewriter>
+  template<typename eT>
+  void SummaryWriter<Filewriter>::Image(const std::string& tag,
+                    int step,
+                    arma::Mat<eT>& matrix,
+                    mlpack::data::ImageInfo& info,
+                    Filewriter& fw,
+                    const std::string& displayName,
+                    const std::string& description)
+
+{
+  // Create a temp directry 
+  int check = mkdir("_tempimage_",0777);
+
+  // Create a vector of temp file names
+  std::vector<std::string> fileNames(matrix.n_cols);
+  for (size_t i = 0; i < matrix.n_cols; i++)
+    fileNames[i] = "_tempimage_/"+std::to_string(i)+".png";
+
+  mlpack::data::Save(fileNames, matrix, info, false);
+
+  std::vector<std::string> encodedImages;
+  mlboard::util::EncodeImage(fileNames, encodedImages);
+
+  Image(tag, step, encodedImages , info.Height(),
+    info.Width(), fw, displayName, description);
+  // Remove all the files
+  for (size_t i = 0; i < matrix.n_cols; i++)
+    remove(fileNames[i].c_str());
+
+  // remove the temp directory
+  rmdir("_tempimage_");
```

Comment:
I tested this code in windows and seem it is working

The only change is `mkdir("_tempimage_",0777)` -> `mkdir("_tempimage_")` since we don't have that definition in windows

Also we can shift `rmdir("_tempimage_")` to `Close()` in filewriter class just like we did for catch test file and I think so that will be ok and will workout in every Operating system

Note: I just tested the bare model Like the logic and not the actual code since I used my friends laptop via remote control :)